lib/queue.js

```diff
@@ -147,16 +147,18 @@ const Queue = function Queue(name, url, opts) {
     client.on('error', this.emit.bind(this, 'error'));
 
     if (type === 'client') {
-      this._initializing = commands(client).then(
-        () => {
-          debuglog(name + ' queue ready');
-        },
-        err => {
-          this.emit('error', new Error('Error initializing Lua scripts'));
-          throw err;
-        }
-      );
-
+      function initScripts(instance) {
+        return commands(client).then(
+            () => {
+              debuglog(name + ' queue ready');
+            },
+            err => {
+              instance.emit('error', new Error('Error initializing Lua scripts'));
+              instance._initializing = initScripts(instance);
+            }
+        );
+      }
+      this._initializing = initScripts(this);
       this._initializing.catch((/*err*/) => {});
```

Comment:
Maybe remove the "instance._initializing = initScripts(instance);" from the error handling part of initscripts, and 
add it to the catch instead.

Maybe the "instance.emit('error', new Error('Error initializing Lua scripts'));" also could be handled in the catch part.

this._initializing.catch((/*err*/) => {
   instance.emit('error', new Error('Error initializing Lua scripts')); // maybe use err here..
   setImmediate(()=> {
      instance._initializing = initScripts(instance);
   });
});